kitsune/products/models/pages.py

```diff
@@ -0,0 +1,154 @@
+
+
+class SearchBlock(blocks.StructBlock):
+    """Block for the search form"""
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        return context
+
+    class Meta:
+        template = "products/blocks/search_block.html"
+        icon = "search"
+        label = "Search Form"
+
+
+class CTABlock(blocks.StructBlock):
+    """Block for the call to action"""
+
+    # Doesn't do much at the moment...#todo
+
+    headline = blocks.CharBlock(required=True, max_length=255)
+    details = blocks.RichTextBlock(required=True)
+    link = blocks.URLBlock(required=True)
+    type = blocks.ChoiceBlock(
+        choices=[
+            ("Community", "Community"),
+            ("Paid", "Paid"),
+            ("Other", "Other"),
+        ]
+    )
+
+    class Meta:
+        template = "products/blocks/cta_block.html"
+        icon = "plus-inverse"
+        label = "Call to Action"
+
+
+class FeaturedArticlesBlock(blocks.StructBlock):
+    """Block for the featured articles"""
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        context["featured"] = get_featured_articles(product=context["product"], locale="en-US")
+        return context
+
+    class Meta:
+        template = "products/blocks/featured_articles_block.html"
+        icon = "doc-full-inverse"
+        label = "Featured Articles"
+
+
+class FeaturedArticleBlock(blocks.StructBlock):
+    """Block for picking a single featured article"""
+
+    article = SnippetChooserBlock(target_model="wiki.Document", required=True)
+
+    class Meta:
+        template = "products/blocks/featured_article_block.html"
+        icon = "doc-full-inverse"
+        label = "Featured Article"
+
+
+class FrequentTopicsBlock(blocks.StructBlock):
+    """Block for the frequent topics"""
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        context["topics"] = topics_for(
+            user=context["user"], product=context["product"], parent=None
+        )
+        return context
+
+    class Meta:
+        template = "products/blocks/frequent_topics_block.html"
+        icon = "doc-full-inverse"
+        label = "Frequent Topics"
+        max = 1
+
+
+class SingleProductIndexPage(Page):
+    """A page representing a product"""
+
+    template = "products/product_wagtail.html"
+
+    product = models.ForeignKey(Product, on_delete=models.PROTECT, related_name="product_index")
+
+    body = StreamField(
+        [
+            ("search", SearchBlock()),
+            ("cta", CTABlock()),
+            ("featured_articles", FeaturedArticlesBlock()),
+            ("featured_article", FeaturedArticleBlock()),
+            ("frequent_topics", FrequentTopicsBlock()),
+            ("text", blocks.RichTextBlock(search_index=True, editor="default")),
```

Comment:
I suspect the default editor is `default` so we don't need to set that. Moreover we don't have a policy yet about search and wagtail and I would avoid in the block definition to set the search index to true

Reply:
Sounds good.